core/src/parser/mod.rs

```diff
@@ -157,7 +161,7 @@ impl MainParser {
 }
 
 /// Parses and returns a Unimarkup document.
-pub fn parse_unimarkup(um_content: &str, config: &mut Config) -> Result<Document, MappedLogId> {
+pub fn parse_unimarkup(um_content: &str, config: &mut Config) -> Document {
```

Comment:
I guess this might be fallible in the future (e.g. invalid macro syntax or something like that). But we should probably think about that and how should we go about it. We might as well always succeed parsing (everything can fallback to paragraph), but pack diagnostics information in `Document` struct.

Reply:
I am also not sure. It would probably make sense to fail if attribute or logic syntax is wrong, and fallback to paragraph for other elements.
If we get to a point where failing makes sense, we can change it back to `Result<>`.